contracts/src/v0.8/functions/dev/1_0_0/FunctionsRouter.sol

```diff
@@ -110,11 +102,14 @@ contract FunctionsRouter is RouterBase, IFunctionsRouter, FunctionsSubscriptions
    *  - adminFee: fee that will be paid to the Router owner for operating the network
    */
   function _updateConfig(bytes memory config) internal override {
-    (uint96 adminFee, bytes4 handleOracleFulfillmentSelector, uint32[] memory maxCallbackGasLimits) = abi.decode(
-      config,
-      (uint96, bytes4, uint32[])
-    );
-    s_config = Config({
+    (
+      uint16 maxConsumers,
+      uint96 adminFee,
+      bytes4 handleOracleFulfillmentSelector,
+      uint32[] memory maxCallbackGasLimits
+    ) = abi.decode(config, (uint16, uint96, bytes4, uint32[]));
+    s_config = SubscriptionConfig({
+      maxConsumers: maxConsumers,
```

Comment:
Maybe we should enforce a reasonable ceiling.. like the 3k number we had

Reply:
It feels like allowlist getter. We don't enforce a limit there but there probably is a point where it will break. Similarly here, it's on us to set a reasonable value and 65k type doesn't seem to outrageous.

Reply:
Setting an arbitrary limit doesn't seem right.  Any limit we set has to be tested first.  I want to hold off unless we are confident in a max value that we know won't break things, which requires investigation time we don't have.

Reply:
@justinkaseman can you explain the risk again?

Reply:
Risk is a subscription has too many consumers and then users cannot remove consumers or cancel their subscription and withdraw funds.

Reply:
Allowlist getter isn't a risk because it is only expected to be used off-chain.

What we are setting here with maxConsumer _is_ used on-chain. It's the upper bound to a loop that we do when we check if a request is in-flight for a subscription.

I think we are okay to go ahead without a ceiling, but did want us to acknowledge it